src/player.cpp

```diff
@@ -6095,9 +6095,9 @@ void player::suffer()
             plut_trans = 0;
             if (tank_plut > 0) {
                 if (has_active_bionic( bio_plut_filter ) ) {
-                    plut_trans = (tank_plut * 0.025);
+                    plut_trans = ( int )( tank_plut * 0.025 );
```

Comment:
What's with all the pointless casting? The assignment will automatically cast the value anyway. And if you absolutely have to cast, why use a potentially dangerous C-style cast instead of a proper `static_cast<>`?

The same applies to the casts in return statements - those casts are automatically done.

Reply:
They're getting flagged because "X type may not fit into X", while explicitly stating the cast informs everything that yes, we realize and accounted for that.

Reply:
> They're getting flagged because "X type may not fit into X", while explicitly stating the cast informs everything that yes, we realize and accounted for that.

For me those casts look like "If I put the cast there, the warning disappears. Don't know why, but it works." In other words: people tend to make those casts *just* to remove the warnings, without actually thinking about the issue.

It also can lead to subtle bugs, like the one with the temperature. And when the type of the assigned variable variable changes, we actually loose accuracy without any reason, e.g. when `int f; ... ; f = (int) 1.5;` changes to `double f;`, it will still hold only 1, not the most likely desired 1.5.

And finally: the cast is done by the anyway compiler - why do the compilers work manually?

Reply:
I guess I've always seen it (on this project at least) as "we've got a lot of contributors who know very little about computer programming, and this not only squashes a potential warning, but also maybe makes them think about what is happening here instead of wondering why they are only getting back integers."

I mean I can x out those few lines if we want, no skin off my back.